File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/java/io/github/lishangbu/avalon/dataset/service/BerryFlavorService.java
package io.github.lishangbu.avalon.dataset.service;

import io.github.lishangbu.avalon.dataset.entity.BerryFlavor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/// 树果风味服务。
public interface BerryFlavorService {

    /// 根据条件分页查询树果风味。
    Page<BerryFlavor> getPageByCondition(BerryFlavor berryFlavor, Pageable pageable);

    /// 新增树果风味。
    BerryFlavor save(BerryFlavor berryFlavor);

    /// 更新树果风味。
    BerryFlavor update(BerryFlavor berryFlavor);

    /// 根据主键删除树果风味。
    void removeById(Long id);
}

File: avalon-modules/avalon-dataset/avalon-dataset-service/src/main/kotlin/io/github/lishangbu/avalon/dataset/service/BerryFlavorService.kt
package io.github.lishangbu.avalon.dataset.service

import io.github.lishangbu.avalon.dataset.entity.dto.BerryFlavorSpecification
import io.github.lishangbu.avalon.dataset.entity.dto.BerryFlavorView
import io.github.lishangbu.avalon.dataset.entity.dto.SaveBerryFlavorInput
import io.github.lishangbu.avalon.dataset.entity.dto.UpdateBerryFlavorInput

/** 树果风味服务 */
interface BerryFlavorService {
    /** 保存树果风味 */
    fun save(command: SaveBerryFlavorInput): BerryFlavorView

    /** 更新树果风味 */
    fun update(command: UpdateBerryFlavorInput): BerryFlavorView

    /** 按 ID 删除树果风味 */
    fun removeById(id: Long)

    /** 按筛选条件查询树果风味列表 */
    fun listByCondition(specification: BerryFlavorSpecification): List<BerryFlavorView>
}
